tests/baselines/reference/objectLiteralNormalization.errors.txt

```diff
@@ -41,10 +39,9 @@ objectLiteralNormalization.ts(49,14): error TS2322: Type '2' is not assignable t
     a2 = { a: "def" };
     a2 = {};
     a2 = { a: "def", b: 20 };  // Error
-    ~~
-!!! error TS2322: Type '{ a: string; b: number; }' is not assignable to type '{ a: number; b: number; } | { a: string; b?: undefined; } | { a?: undefined; b?: undefined; }'.
-!!! error TS2322:   Types of property 'b' are incompatible.
-!!! error TS2322:     Type 'number' is not assignable to type 'undefined'.
+                     ~
+!!! error TS2322: Type 'number' is not assignable to type 'undefined'.
+!!! related TS6500 objectLiteralNormalization.ts:11:19: The expected type comes from property 'b' which is declared here on type '{ a: number; b: number; } | { a: string; b?: undefined; } | { a?: undefined; b?: undefined; }'

```

Comment:
Unlike the other regressions, it really feels like we've lost too much context here.

Reply:
Doesn't the related info contain the same information that was reported previously in the non-elementwise error? What would be your ideal outcome here? It could be nicer if the error would focus on the selected best type and not on the containing union. 

I wouldn't call this bit a regression per se - the same information is still here, it's presented differently though. And, of course, the presentation **does** matter - I'm not claiming that it doesn't.

Reply:
FWIW: That error is confusing.  It starts out saying `number` isn't assignable to `undefined`, which is true, but then goes on to claim that the "expected type" comes from `b` of `{ a: number; b: number; } | { a: string; b?: undefined; } | { a?: undefined; b?: undefined; }`.  Now, given that union as the target type, if the `b` in the source is the only problem... why *can't* it be a number?  So the error message kinda-sorta contradicts itself.

Reply:
It makes sense. When rereading it now I see how this is confusing - perhaps the related information should mention the complete source type or it should somehow mention the selected best type based on which this elementwise elaboration is produced.

I'll look into improving this - thanks for the feedback ❤️